backend/src/main/java/de/neuefische/backend/security/SecurityConfig.java

```diff
@@ -0,0 +1,40 @@
+package de.neuefische.backend.security;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.HttpStatusEntryPoint;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+    @Value("${app.url}")
+    private String appUrl;
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+
+        http
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(a -> a
+                        .requestMatchers("/api/auth/me").authenticated()
+                        .requestMatchers("/api/chat").authenticated()
+                        .requestMatchers(HttpMethod.POST).authenticated()
+                        .requestMatchers(HttpMethod.DELETE).authenticated()
+                        .requestMatchers(HttpMethod.PUT).authenticated()
+                        .anyRequest().permitAll()
```

Comment:
you could do white listing instead of black listing here. anyRequest can be authenticated and you say with permitAll which routes are allowed from anyone. Think of a bigger project where you may have houndreds of routes and you forgot some route to secure which is very important. Using white listing would be a better choice :)

Reply:
Very good point! I'm just not sure, how to do it. I tried the following, but then I won't be able to login.. 
 http
                .csrf(AbstractHttpConfigurer::disable)
                .authorizeHttpRequests(a -> a
                        .requestMatchers(HttpMethod.GET, "/", "/api/recipes", "/api/auth/me").permitAll()
                        .anyRequest().authenticated()
                )
                
                it would be grate, if you could give me a hint. Thx!!

Reply:
Great point! I hope, I did it correctly 😄